bundles/config/org.eclipse.smarthome.config.discovery.usbserial.linux.sysfs/src/main/java/org/eclipse/smarthome/config/discovery/usbserial/linux/sysfs/internal/UsbSerialScanner.java

```diff
@@ -0,0 +1,38 @@
+/**
+ * Copyright (c) 2014,2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.smarthome.config.discovery.usbserial.linux.sysfs.internal;
+
+import java.util.Set;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.smarthome.config.discovery.usbserial.UsbSerialDeviceInformation;
+
+/**
+ * Implementations of this interface scan for serial ports provided by USB devices.
+ *
+ * @author Henning Sudbrock - initial contribution
+ */
+@NonNullByDefault
+public interface UsbSerialScanner {
+
+    /**
+     * Performs a single scan for serial ports provided by USB devices.
+     *
+     * @return A collection containing all scan results.
+     * @throws Exception any checked exception that prevented the scan. Note that exceptions preventing the successful
+     *             identification of a single USB device might be swallowed, simply skipping that device while still
+     *             discovering other devices successfully.
+     */
+    Set<UsbSerialDeviceInformation> scan() throws Exception;
```

Comment:
I believe it would be really helpful for consumers to handle a specific exception than a generic one. We could therefore introduce a custom checked exception.

Reply:
@amitjoy The reason I chose to throw an `Exception` here (and not a custom exception) is that this interface is only used internally (it is in an internal package, and only used by classes in this internal package). Hence, there will be no other consumers of this interface.

Do you think that a specific exception should be thrown here nevertheless?

Reply:
It should not depend if it is used internally only or externally.
Shouldn't an I/O exception fit your needs?

Reply:
For the single implementation of `UsbSerialScanner` (the `SysfsUsbSerialScanner`) an `IOException` is indeed the suitable exception. Thanks for that suggestion, @maggu2810.

I will change the code accordingly.

In case there will be other implementations of `UsbSerialScanner` in the future that need other exception types, we could still think about a custom `UsbSerialScanningException` then...